time/src/test/java/net/jqwik/time/api/LocalDateTimeTests.java

```diff
@@ -0,0 +1,349 @@
+			LocalDateTime max = LocalDateTime.of(date.plusDays(1), LocalTime.of(0, 0, secondStart));
+
+			LocalDateTimeArbitrary dateTimes = DateTimes.dateTimes().between(min, max);
+
+			assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+				assertThat(dateTime).isNotNull();
+				return true;
+			});
+
+		}
+
+	}
+
+	@Group
+	class SimpleAnnotations {
+
+		@Property
+		void validLocalDateTimeIsGenerated(@ForAll LocalDateTime dateTime) {
+			assertThat(dateTime).isNotNull();
+		}
+
+	}
+
+	@Group
+	class CheckDateTimeMethods {
+
+		@Group
+		class DateTimeMethods {
+
+			@Property
+			void atTheEarliest(@ForAll("dateTimes") LocalDateTime min, @ForAll Random random) {
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().atTheEarliest(min);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isAfterOrEqualTo(min);
+					return true;
+				});
+
+			}
+
+			@Property
+			void atTheLatest(@ForAll("dateTimes") LocalDateTime max, @ForAll Random random) {
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().atTheLatest(max);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isBeforeOrEqualTo(max);
+					return true;
+				});
+
+			}
+
+			@Property
+			void between(@ForAll("dateTimes") LocalDateTime min, @ForAll("dateTimes") LocalDateTime max, @ForAll Random random) {
+
+				Assume.that(!min.isAfter(max));
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(min, max);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isAfterOrEqualTo(min);
+					assertThat(dateTime).isBeforeOrEqualTo(max);
+					return true;
+				});
+
+			}
+
+			@Property
+			void betweenMinAfterMax(@ForAll("dateTimes") LocalDateTime min, @ForAll("dateTimes") LocalDateTime max, @ForAll Random random) {
+
+				Assume.that(min.isAfter(max));
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(min, max);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isAfterOrEqualTo(max);
+					assertThat(dateTime).isBeforeOrEqualTo(min);
+					return true;
+				});
+
+			}
+
+			@Property
+			void betweenSame(@ForAll("dateTimes") LocalDateTime same, @ForAll Random random) {
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(same, same);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isEqualTo(same);
+					return true;
+				});
+
+			}
+
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			LocalDateTimeArbitrary dateTimes = DateTimes.dateTimes();
+			LocalDateTime value = falsifyThenShrink(dateTimes, random);
+			assertThat(value).isEqualTo(LocalDateTime.of(1900, Month.JANUARY, 1, 0, 0, 0));
+		}
+
+		@Disabled("Not working at the moment.")
```

Comment:
Fixing that will probably require some additional functionality in jqwik core. Enable with current best shrink.